service/src/main/resources/api/openapi.yml

```diff
@@ -35,9 +35,56 @@ paths:
               schema:
                 $ref: '#/components/schemas/VersionProperties'
 
+  # User profile management
+  /api/profile:
+    parameters:
+      - $ref: '#/components/parameters/ProfilePath'
+    get:
+      summary: Retrieve a profile property.
+      operationId: userProfileLookup
+      tags: [ Profile ]
+      responses:
+        '200':
+          $ref: '#/components/responses/LookupResult'
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '403':
+          $ref: '#/components/responses/PermissionDenied'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '500':
+          $ref: '#/components/responses/ServerError'
+    put:
+      summary: Set a profile property.
+      operationId: userProfileSet
+      tags: [ Profile ]
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/AnyObject'
+      responses:
+        '204':
```

Comment:
What if we just return the object is the "put" method as well (in the same format as "get")?

Seems like a pretty standard thing to do.  Don't have a great use case, but the caller may want to use the result object in the same way as a read, without building it themselves?